test/k8s_service_example_test_helpers.go

```diff
@@ -176,3 +182,58 @@ func verifyServiceAvailable(
 		validationFunction,
 	)
 }
+
+// verifyServiceRoutesToMainAndCanaryPods ensures that the service is routing to both the main and the canary pods
+// It does this by repeatedly issuing requests to the service and inspecting the nginx Server header
+// Once both nginx tags have been seen in this header - we can be confident that we've reached both types of pod via the service
+func verifyServiceRoutesToMainAndCanaryPods(
+	t *testing.T,
+	kubectlOptions *k8s.KubectlOptions,
+	appName string,
+	releaseName string,
+) {
+
+	// Get the service and wait until it is available
+	filters := metav1.ListOptions{
+		LabelSelector: fmt.Sprintf("app.kubernetes.io/name=%s,app.kubernetes.io/instance=%s", appName, releaseName),
+	}
+	services := k8s.ListServices(t, kubectlOptions, filters)
+	require.Equal(t, len(services), 1)
+	service := services[0]
+	k8s.WaitUntilServiceAvailable(t, kubectlOptions, service.Name, WaitTimerRetries, WaitTimerSleep)
+
+	service = *k8s.GetService(t, kubectlOptions, service.Name)
```

Comment:
NIT: It might be safer to use a new var here to avoid dereferencing the pointer.

Reply:
Used a new variable in 838299c